app/controllers/api/v2/host_classes_controller.rb

```diff
@@ -36,7 +36,7 @@ def destroy
       # overwrite resource_name so it's host and and not host_class, since we want to return @host
       def find_host
         not_found and return false if params[:host_id].blank?
-        @host = Host.find(params[:host_id]) if Host::Managed.respond_to?(:authorized) &&
+        @host = Host.friendly.find(params[:host_id]) if Host::Managed.respond_to?(:authorized) &&
```

Comment:
the logic in host classes is slightly different from the logic in find common, I didn't change that, just added the friendly (from the horrible friendly_id gem)